packages/cubejs-server-core/src/core/OrchestratorApi.ts

```diff
@@ -1,20 +1,25 @@
 /* eslint-disable no-throw-literal */
-const pt = require('promise-timeout');
-const { QueryOrchestrator, ContinueWaitError } = require('@cubejs-backend/query-orchestrator');
+import pt from 'promise-timeout';
+import { QueryOrchestrator, ContinueWaitError } from '@cubejs-backend/query-orchestrator';
 
 export class OrchestratorApi {
-  constructor(driverFactory, logger, options) {
-    options = options || {};
-    this.options = options;
+  public seenDataSources: { [dataSource: string]: boolean };
```

Comment:
Let's make it private, because we can modify it by public method - `addDataSeenSource `